image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -84,12 +84,14 @@
     {{- if kindIs "invalid" $._rox.scannerV4.indexer.serviceTLS.generate }}
       {{/* We need special handling here, because 'generate' will default to 'false' on upgrades. */}}
       {{/* And in case scanner V4 was not deployed earlier, we need to make sure that these resources are correctly initialized. */}}
-      {{- $lookupOut := dict -}}
-      {{- $_ := include "srox.safeLookup" (list $ $lookupOut "v1" "Secret" $.Release.Namespace "scanner-v4-indexer-tls") -}}
-      {{- if not $lookupOut.result -}}
-        {{/* If generate=null and the resource does not exist yet, attempt to create it. */}}
-        {{/* If lookup is not possible (e.g. in the operator), then 'generate' needs to be set correctly. */}}
-        {{- $_ := set $._rox.scannerV4.indexer.serviceTLS "generate" true -}}
+      {{- if $.Release.IsUpgrade -}}
+        {{- $lookupOut := dict -}}
+        {{- $_ := include "srox.safeLookup" (list $ $lookupOut "v1" "Secret" $.Release.Namespace "scanner-v4-indexer-tls") -}}
+        {{- if not $lookupOut.result -}}
+          {{/* If generate=null and the resource does not exist yet, attempt to create it. */}}
+          {{/* If lookup is not possible (e.g. in the operator), then 'generate' needs to be set correctly. */}}
```

Comment:
Not in scope of the PR, but for my curiosity - what does "needs to be set correctly" mean? This code needs to set it, or some other place needs to take care of this? And what does "correctly" mean?
IIRC the operator disables helm chart's TLS cert generation logic anyway, so why does the value of "generate" matter in this case?

Reply:
Thanks for asking, allows me to sort my vague thoughts on this.

What I mean here is this:

Since the strategy is to pull in Scanner V4 automatically not only on fresh installs, but also on upgrades, the required secrets for Scanner V4 components need to be created. The logic applied so far in the Helm chart (IsInstall == false => generate := false) is not sufficient anymore in this case.

The comment above refers to the fact that the operator sets these `generate` flags explicitly, hence we cannot depend on this kind of "auto-sensing" within the chart.
In the end, the secrets for Scanner V4 must be there. How do they get created?

So, as I see it, either the operator needs to
(1) keep the `generate := false` and then take care of creating these secrets
or
(2) set `generate := true` and then let the Helm chart take care of creating these certificates (which would require that the CA cert + key is passed to the Helm chart rendering?).

I am not familiar with how this works currently.

FYI @johannes94

Reply:
We definitely want (1), consistent with the other secrets.